source/ScreensaverHero.hx

```diff
@@ -6,40 +6,50 @@ import flixel.util.FlxSpriteUtil;
 import flixel.math.FlxVelocity;
 
 class ScreensaverHero extends FlxSprite {
-    private static var maxHexColor = 16777216;
+    private static var SPRITE_WIDTH(default, null):Int = 128;
+    private static var SPRITE_HEIGHT(default, null):Int = 64;
 
-    private var running = false;
+    private static var MOUSE_FLEE_DISTANCE(default, null):Int = 100;
+    private static var MOUSE_FLEE_SPEED(default, null):Int = 20;
+    private static var MOUSE_FLEE_TIMEOUT_MS(default, null):Int = 1000;
+
+    private static var MIN_HEX_COLOR(default, null) = 0;
+    private static var MAX_HEX_COLOR(default, null) = 16777216;
 
     var velY = FlxG.random.int(15,25) * FlxG.random.sign();
     var velX = FlxG.random.int(15,25) * FlxG.random.sign();
 
-    public function new()
+    public function new(x=0,y=0)
     {
         super();
-        makeGraphic(128, 64);
-        screenCenter();
-        this.y -= 50;
+        makeGraphic(SPRITE_WIDTH, SPRITE_HEIGHT);
+        this.x = x;
+        this.y = y;
 
         this.velocity.x = velY;
         this.velocity.y = velX;
 
-        this.color = FlxG.random.int(0, maxHexColor);
+        this.color = FlxG.random.int(MIN_HEX_COLOR, MAX_HEX_COLOR);
     }
 
     override public function update(elapsed:Float)
     {
         super.update(elapsed);
         FlxSpriteUtil.screenWrap(this);
-        this.color += Std.int(4*(this.x + this.y)) % maxHexColor;
 
-        if (getDistanceFromMouse(FlxG.mouse) < 100) {
-            FlxVelocity.moveTowardsMouse(this, 20, 1000);
+        // Sign will make it randomly increase OR decrease by that amount
+        var newColor = (this.color + Std.int(4*(this.x + this.y))) * FlxG.random.sign();    
```

Comment:
This does make the color change every frame, but I don't think it works the way we actually want - the `FlxG.random.sign()` is making the color fully swap between negative and positive values (instead of incrementing by positive and negative values). A negative color just makes the whole sprite completely black, which makes the rectangle flash between black and blue constantly.

It's the kind of strobing effect that I think you'd have to warn people about, I think it'd be better to fix this logic so it's not in the final version of this code.

there are lots of ways to help it not do this flashing thing - just removing the `* FlxG.random.sign()` would do the trick.

Reply:
I think it's also worth taking a hard look at how you're building this color - on some level, it's not that important because it's just a cosmetic color in a basic assignment. On some level, though, it's representative of how we'll do problem-solving when making game logic.

It's very worthwhile for us to practice this process, where we decide what we want to see and then determine what sort of code and method calls will get us the result that we want.